iocore/net/UnixNetVConnection.cc

```diff
@@ -1138,28 +1133,22 @@ UnixNetVConnection::acceptEvent(int event, Event *e)
     return EVENT_DONE;
   }
 
-  SET_HANDLER((NetVConnHandler)&UnixNetVConnection::mainEvent);
-
-  PollDescriptor *pd = get_PollDescriptor(thread);
-  if (ep.start(pd, this, EVENTIO_READ | EVENTIO_WRITE) < 0) {
-    Debug("iocore_net", "acceptEvent : failed EventIO::start");
+  // Send this NetVC to NetHandler and start to polling read & write event.
+  if (h->startIO(this) < 0) {
     free(t);
     return EVENT_DONE;
   }
 
-  nh = get_NetHandler(thread);
-  set_inactivity_timeout(0);
-  nh->open_list.enqueue(this);
+  // Setup a timeout callback handler.
+  SET_HANDLER((NetVConnHandler)&UnixNetVConnection::mainEvent);
```

Comment:
Any reason this was moved after  `startIO` instead of before? Seems more robust to be before in case of immediately I/O.

Reply:
The ```UnixNetVConnection::mainEvent``` is designed to handle timeout callback.
ATS should setup timeout handler for a NetVC and then push the netvc into InactivityCop.
The ```open_list``` is a monitor list, all NetVCs inside of it will be checked by InactivityCop.

It is more robust to move it after ```startIO```.

Reply:
OK. I suppose it also doesn't matter because the net handler lock is held at this point so no I/O will dispatch.